direct/io-bulk-fs/src/main/java/cz/o2/proxima/direct/bulk/AbstractBulkFileSystemAttributeWriter.java

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.direct.bulk;
+
+import com.google.common.annotations.VisibleForTesting;
+import cz.o2.proxima.direct.core.AbstractBulkAttributeWriter;
+import cz.o2.proxima.direct.core.BulkAttributeWriter;
+import cz.o2.proxima.direct.core.CommitCallback;
+import cz.o2.proxima.direct.core.Context;
+import cz.o2.proxima.functional.Factory;
+import cz.o2.proxima.repository.EntityDescriptor;
+import cz.o2.proxima.storage.StreamElement;
+import cz.o2.proxima.util.ExceptionUtils;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+/** An abstract parent class for {@link BulkAttributeWriter}. */
+@Slf4j
+public abstract class AbstractBulkFileSystemAttributeWriter extends AbstractBulkAttributeWriter {
```

Comment:
This is the only implementation of `AbstractBulkAttributeWriter`. Maybe we can merge these?

Reply:
Hm, I think I can move the one from `direct-core` here and merge it. :+1:

Reply:
Hm, this is not the only implementation. There is `InMemStorage.Writer` implementation. Therefore, I'd keep it separated, because I don't want to pull this to core and core cannot depend on this (cyclic dependency).